extensions/interactions/FractionInput/directives/FractionInputValidationService.js

```diff
@@ -88,7 +118,32 @@ oppia.factory('FractionInputValidationService', [
                 var f = toFloat(rule.inputs.f);
                 setLowerAndUpperBounds(range, -Infinity, f, false, false);
                 break;
+              case 'HasNumeratorEqualTo':
+                if (!Number.isInteger(rule.inputs.x)) {
+                  warningsList.push(getNonIntegerInputWarning(i, j));
+                }
+                break;
+              case 'HasIntegerPartEqualTo':
+                if (!Number.isInteger(rule.inputs.x)) {
+                  warningsList.push(getNonIntegerInputWarning(i, j));
+                }
+                break;
+              case 'HasDenominatorEqualTo':
+                if (!Number.isInteger(rule.inputs.x)) {
+                  warningsList.push(getNonIntegerInputWarning(i, j));
+                }
+                if (rule.inputs.x == 0) {
+                  warningsList.push({
+                    type: WARNING_TYPES.ERROR,
+                    message: (
+                      'Rule ' + (j + 1) + ' from answer group ' +
+                      (i + 1) + ' is invalid as denominator ' +
```

Comment:
...is invalid: the denominator should be greater than zero.

Reply:
Done.